### DISCUSSION TITLE
Missing ExtractMode in ExcelEmitter

### DISCUSSION THREAD

**Discussion by raoulduke98** [eclipse-birt/birt#1279] [Q&A] [answered]
I am just upgrading from BIRT Runtime 4.3.2. to 4.13.0 and noticed that the "ExtractMode" mode is no longer supported by the integrated Spudsoft ExcelEmitter, which makes use of the SXSSFWorkbook. Is there a specific reason for this or is there a chance of getting the feature back? Or is there an alternative emitter that supports streaming of the XLSX format?

**Comment by hvbtup** [comment, marked as answer]
That feature was never a part of BIRT.

10 years ago, Jim Talbut donated the "Spudsoft Excel Emitter" to the BIRT project, see the Git history of `ExcelEmitter.java`.

After that, he improved the emitter. But that was only on his BitBucket repository.
If you wanted to use any of the new or improvied features (the ExtractMode being one of them), you needed to replace the plugin with his version.

You might try the `tup_main` branch at https://hithub.com/triestram-partner/birt at your own risk.
We used some of the new features, and since they didn't make it into any of the newer BIRT releases, I finally decided to build a customized version of BIRT suited for our needs.

Be warned: 
- The source code under `uk.*` differs significantly from the `master` branch. 
- I also changed a lot in the Word emitter, so the DOCX results will probably look different.
- We didn't use the ExtractMode feature.

You may consider creating your own fork of the https://github.com/eclipe/birt, then create your own branch, and more or less copying the uk.* code from my tup_main branch into your branch.

However, I don't see an easy and safe way to merge all those changes into the master branch, because I am not sure that there are no breaking changes.

**Comment by raoulduke98** [comment]
I see, thanks for your quick reply!
